module/zstd/common/zstd_common.c

```diff
@@ -0,0 +1,47 @@
+/* BEGIN CSTYLED */
+/*
+ * Copyright (c) 2016-present, Yann Collet, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under both the BSD-style license (found in the
+ * LICENSE file in the root directory of this source tree) and the GPLv2 (found
+ * in the COPYING file in the root directory of this source tree).
+ * You may select, at your option, one of the above-listed licenses.
+ */
+
+
+
+/*-*************************************
+*  Dependencies
+***************************************/
+#include <sys/zstd/mem.h>
+#include <sys/zstd/error_private.h>
+#include <sys/zstd/zstd_internal.h>
+
+/*! g_debuglog_enable :
+ *  turn on/off debug traces (global switch) */
+#if defined(ZSTD_DEBUG) && (ZSTD_DEBUG >= 2)
+int g_debuglog_enable = 1;
+#endif
+
+
+/*=**************************************************************
+*  Custom allocator
+****************************************************************/
+void* ZSTD_malloc(size_t size, ZSTD_customMem customMem)
+{
+    if (customMem.customAlloc)
+        return customMem.customAlloc(customMem.opaque, size);
+    return malloc(size);
```

Comment:
I don't think it's a good idea to add a `malloc()` wrapper for this, let's just `kmem_alloc()` which will work in both user and kernel space due to existing macros.

Reply:
this code will not be reached. its just from the original zstd code implementatoin. customAlloc is always set. of course i can remove it. but i want to keep the code original which allows easier merging from newer zstd code.